kotlinpoet/src/commonTest/kotlin/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -596,6 +596,47 @@ class TypeSpecTest {
     )
   }
 
+  @Test fun enumWithPrimaryConstructorAndMultipleInterfaces() {
+    val roshambo = TypeSpec.enumBuilder("Roshambo")
+      .addSuperinterface(Runnable::class)
+      .addSuperinterface(Cloneable::class)
+      .addEnumConstant(
+        "SCISSORS",
+        TypeSpec.anonymousClassBuilder()
+          .addSuperclassConstructorParameter("%S", "peace sign")
+          .build(),
+      )
+      .addProperty(
+        PropertySpec.builder("handPosition", String::class, KModifier.PRIVATE)
+          .initializer("handPosition")
+          .build(),
+      )
+      .primaryConstructor(
+        FunSpec.constructorBuilder()
+          .addParameter("handPosition", String::class)
+          .build(),
+      )
+      .build()
+    assertThat(toString(roshambo)).isEqualTo(
+      """
+        |package com.squareup.tacos
+        |
+        |import java.lang.Runnable
+        |import kotlin.Cloneable
+        |import kotlin.String
+        |
+        |public enum class Roshambo(
+        |  private val handPosition: String,
+        |) : Runnable,
+        |    Cloneable {
+        |  SCISSORS("peace sign"),
```

Comment:
ktlint wants this to be
```suggestion
        |public enum class Roshambo(
        |  private val handPosition: String,
        |) : Runnable,
        |  Cloneable {
        |  SCISSORS("peace sign"),
```
which I absolutely hate. Visually there's no delineation between the supertypes and the body.

We could force a leading newline and do
```suggestion
        |public enum class Roshambo(
        |  private val handPosition: String,
        |) : Runnable,
        |  Cloneable {
        |
        |  SCISSORS("peace sign"),
```
which ktlint allows. It's not great, but I'd be okay with this.

Thoughts? We could also just keep what I've already done. Most people aren't formatting generated code and I prefer mine best.

Reply:
The [code style](https://kotlinlang.org/docs/coding-conventions.html#class-headers) suggests that all supertypes should have the same indentation, so I think it's a bug in ktlint? I also prefer it this way, so let's keep it as is.

Reply:
They only mention horizontal alignment when you make a line break after the colon for non-primary constrictor cases.

The example with a wrapping primary constructor only aligns because they are doing a 4-space indent. So unfortunately I think ktlint is correct in what it does.

That being said, we can still leave it.

Reply:
> The example with a wrapping primary constructor only aligns because they are doing a 4-space indent.

Good point.